src/typing.ml

```diff
@@ -317,11 +341,11 @@ let rec dterm kid crcm ns denv { term_desc; term_loc = loc } : dterm =
       let ls = find_q_ls ns q in
       if ls.ls_field then
         W.error ~loc (W.Symbol_not_found (string_list_of_qualid q));
-      if ls.ls_args <> [] then
-        W.error ~loc (W.Partial_application ls.ls_name.id_str);
-      let _, dty = specialize_ls ls in
-      let node, dty = (DTapp (ls, []), dty) in
-      mk_dterm ~loc node dty
+      if ls.ls_args <> [] then gen_app ~loc ls []
```

Comment:
Maybe it's the opportunity to use a pattern matching here.

Reply:
Thank you for pinpointing that place, it made me realise that distinguishing cases is really not necessary there, `gen_app` will do exactly that special case anyway!